### PR TITLE
Oidc docs

### LINKED ISSUE #2988: Split our OIDC docs per provider
### Description:

https://github.com/kubeapps/kubeapps/pull/2982#discussion_r652514798

As the currents docs are getting becoming more detailed, perhaps it's better to split the docs and just point to the files containing the información per provider. This way we can be as wordy as we want without compromising the reader's experience.

### PR DESCRIPTION
**Pull request by ppbaena**
### Description of the change

Edit project documentation to improve OIDC configuration readability by splitting the different IdP providers in separate files, as well as a separate section for troubleshooting and debugging. Reorder some paragraphs (link to Michael's blog post) to guide users to their final objective within official documentation and then, as extra info users can jump into a complete example.

### Benefits

Improve readability and simplify focusing on different IdP providers according to the user needs.

### Possible drawbacks

Increase the number of documentation files.

### Applicable issues

  - fixes #2988

### Additional information
N/A

### CODE REVIEW

File: docs/user/using-an-OIDC-provider.md [hunk 1]
```diff
@@ -22,444 +22,29 @@ When configuring the identity provider, you will need to ensure that the redirec
...
 
-When using the default auth proxy, some users may experience the behavior where they are automatically logged out from the console.
-Prior to the Kubeapps chart version 7.1.0, the auth proxy configuration did not include a default `--cookie-refresh` value to refresh the access/openid token and so the console will logout once the token expires. In the case of Keycloak for example, this can happen quickly as the default access token expiration is 5m.
+- [VMware Cloud Services](./OAuth2OIDC-VMware-cloud-services.md)
+- [Azure Active Directory](./OAuth2OIDC-azure-active-directory.md)
+- [Google OpenID Connect](./OAuth2OIDC-google-openid-connect.md)
+- [Keycloak](./OAuth2OIDC-keycloak.md)
+- [Dex](OAuth2OIDC-dex.md)
+- [Oauth2-proxy](./OAuth2OIDC-oauth2-proxy.md)
```
Comment (absoludity) [inline docs/user/using-an-OIDC-provider.md:42]: oauth2-proxy doesn't really belong like this as another identity provider - it's not one at all. We use oauth2-proxy to proxy oauth2 calls to an identity provider. I'd probably leave the oauth2-proxy details in the main file still?

**Review by absoludity — APPROVED** [review]
[Rendered version can be viewed in your branch here](https://github.com/ppbaena/kubeapps/blob/OIDC-docs/docs/user/using-an-OIDC-provider.md). Often handy for reviewers to include that in the PR description.

Looks much cleaner. I'd just keep the oauth2-proxy details in the main file as it's the default auth proxy that we run with and link to the keycloak file that you've created as an alternative. But if you want to keep them both as separate files, perhaps just clearly identify them as different from the identity providers that you list.

File: docs/user/OAuth2OIDC-VMware-cloud-services.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# VMware Cloud Services
```
Comment (antgamdia) [inline docs/user/OAuth2OIDC-VMware-cloud-services.md:1]: I'd change the title to "Configuring XXX as an OIDC provider" (using just the provider name as the title could be misleading, no?

File: docs/user/OAuth2OIDC-VMware-cloud-services.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
```
Comment (antgamdia) [inline docs/user/OAuth2OIDC-VMware-cloud-services.md:2]: I'd add sth like: "this document explains how con configure XXX as an OIDC provider, check out the [parent document](./using-an-OIDC-provider.md) if you are looking for more general information"

File: docs/user/OAuth2OIDC-azure-active-directory.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
+# Azure Active Directory
```
Comment (antgamdia) [inline docs/user/OAuth2OIDC-azure-active-directory.md:1]: ditto

File: docs/user/OAuth2OIDC-dex.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# Dex
```
Comment (antgamdia) [inline docs/user/OAuth2OIDC-dex.md:1]: ditto

File: docs/user/OAuth2OIDC-google-openid-connect.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# Google OIDC
```
Comment (antgamdia) [inline docs/user/OAuth2OIDC-google-openid-connect.md:1]: ditto

File: docs/user/OAuth2OIDC-keycloak.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+# Keycloak
```
Comment (antgamdia) [inline docs/user/OAuth2OIDC-keycloak.md:1]: Don't we have a comprehensive guide written by @dlaloue-vmware  here?

File: docs/user/OAuth2OIDC-keycloak.md [hunk 8]
```diff
@@ -0,0 +1,7 @@
...
+
```
Comment (antgamdia) [inline docs/user/OAuth2OIDC-keycloak.md:2]: ditto

**Review by antgamdia — APPROVED** [review]
Thanks for the changer here! It's great to have it finally split into different pieces.
Should we place them into a different subfolder (so that the rest of top-level docs remain separated from these ones?)

File: docs/user/OAuth2OIDC-keycloak.md (hunk 8 above)
Comment (ppbaena) [reply docs/user/OAuth2OIDC-keycloak.md:2]: Agree, let me reorder that and include some extra info to the headline so if you land directly to that page, you can get some context.

**Pull request merged**